contracts/features-staking/StakingCampaignRecipient.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '../V2/NonCompoundingRewardsPool.sol';
+
+/** @dev Staking pool with a time lock and throttled exit. 
+    Inherits all staking logic from RewardsPoolBase.
+    Only allows exit at the end of the time lock and via the throttling mechanism.
+*/
+contract StakingCampaignRecipient is NonCompoundingRewardsPool {
+    using SafeERC20 for IERC20;
+    mapping(address => uint256[]) public pendingRewardsAmount;
+
+    /** @param _stakingToken The token to stake
+     * @param _rewardsTokens The reward tokens
+     * @param _stakeLimit Maximum amount of tokens that can be staked per user
+     * @param _throttleRoundSeconds Seconds per throttle round
+     * @param _throttleRoundCap Maximum tokens withdrawn per throttle round
+     * @param _contractStakeLimit Maximum amount of tokens that can be staked in total
+     * @param _name Name of the pool
+     */
+    constructor(
+        IERC20 _stakingToken,
+        address[] memory _rewardsTokens,
+        uint256 _stakeLimit,
+        uint256 _throttleRoundSeconds,
+        uint256 _throttleRoundCap,
+        uint256 _contractStakeLimit,
+        string memory _name
+    )
+        NonCompoundingRewardsPool(
+            _stakingToken,
+            _rewardsTokens,
+            _stakeLimit,
+            _throttleRoundSeconds,
+            _throttleRoundCap,
+            _contractStakeLimit,
+            _name
+        )
+    {}
+
+    /**
+     * @dev Add the stakers in a batch to the recipient contract.
+     */
+    function addStakersBatch(
+        address[] calldata _stakers,
+        uint256[] calldata _stakingAmounts,
+        uint256[][] calldata _rewardAmounts
+    ) external onlyOwner {
+        uint256 stakersLength = _stakers.length;
+        uint256 stakingAmountsLength = _stakingAmounts.length;
+
+        require(
+            stakersLength == stakingAmountsLength,
+            'StakingCampaignRecipient: stakers and staking amounts length mismatch'
+        );
+
+        for (uint256 i = 0; i < stakersLength; i++) {
+            _addStaker(_stakingAmounts[i], _stakers[i], _rewardAmounts[i]);
+        }
+    }
+
+    function _addStaker(
+        uint256 _tokenAmount,
+        address _staker,
+        uint256[] calldata _rewardAmount
+    ) internal {
+        uint256 currentTimestamp = block.timestamp;
+        require(
```

Comment:
You can move this require to the addStakersBatch so you only check it once instead of for every user